vapoursynth-sys/build.rs

```diff
@@ -61,20 +66,36 @@ fn get_default_library_dir() -> Option<impl Iterator<Item = String>> {
         "programfiles(x86)"
     }));
 
-    let suffix = if env::var("TARGET").ok()?.starts_with("i686") {
+    let suffix = if target.starts_with("i686") {
         "lib32"
     } else {
         "lib64"
     };
 
-    Some(programfiles.flat_map(move |programfiles| {
-        // Use both VapourSynth and VapourSynth-32 folder names.
-        ["", "-32"].iter().filter_map(move |vapoursynth_suffix| {
-            let mut path = PathBuf::from(&programfiles);
-            path.push(format!("VapourSynth{}", vapoursynth_suffix));
-            path.push("sdk");
-            path.push(suffix);
-            path.to_str().map(|s| s.to_owned())
+    programfiles
+        .flat_map(move |programfiles| {
+            // Use both VapourSynth and VapourSynth-32 folder names.
+            ["", "-32"].iter().filter_map(move |vapoursynth_suffix| {
+                let mut path = PathBuf::from(&programfiles);
+                path.push(format!("VapourSynth{}", vapoursynth_suffix));
+                path.push("sdk");
+                path.push(suffix);
+                path.to_str().map(|s| s.to_owned())
+            })
         })
-    }))
+        .collect()
+}
+
+// Returns the default library dirs on macOS when using homebrew.
+fn get_default_macos_library_dir(target: String, host: String) -> Vec<String> {
+    // If the host is not macOS/Apple, the library dirs will be different.
+    if !host.contains("apple-darwin") {
+        return vec![];
+    }
+
+    if target.starts_with("aarch64") {
+        vec![String::from("/opt/homebrew/lib/")]
+    } else {
+        vec![String::from("/usr/local/homebrew/lib/")]
```

Comment:
If we still want this in, it would probably best to use `HOMEBREW_PREFIX` instead of hardcoding the paths according to the arch.

Reply:
Is `HOMEBREW_PREFIX` an environment variable set by default?

The reason for hardcoding based on target arch was the [FAQ](https://docs.brew.sh/FAQ#why-is-the-default-installation-prefix-opthomebrew-on-apple-silicon):
> **Why is the default installation prefix /opt/homebrew on Apple Silicon?**
> 
> The prefix /opt/homebrew was chosen to allow installations in /opt/homebrew for Apple Silicon and /usr/local for Rosetta 2 to coexist and use bottles.

That way, it should be possible to have libraries for cross-compilation and still have it work reasonably well out of the box. Users with custom prefixes would need to manually specify this.

Reply:
Well, we do know the target architecture at compile-time. I tried accounting for HOMEBREW_PREFIX, how about this?

Reply:
Are you distributing a dynamically linking executable? It should only matter in that situation and it seems uncommon to me.

Reply:
Hm.. I don't see why this would not matter to statically linked executables.

Either way, both av1an (linked in the related issue) and the rust-av projects only use dynamic linking for vapoursynth as far as I'm aware. Don't forget that VS itself is LGPL-licensed, so it would only make sense (IMHO) to support dynamic linking.

Reply:
Right, I forget that VS doesn't seem to ever be built in statically.